sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubServiceAsyncClient.java

```diff
@@ -131,8 +131,13 @@
         if (requestOptions == null) {
             requestOptions = new RequestOptions();
         }
-        requestOptions.addHeader("contentType", contentType.toString());
-        requestOptions.addHeader("contentLength", String.valueOf(contentLength));
+        requestOptions.addHeader("Content-Type", contentType.toString());
+        requestOptions.addRequestCallback(httpRequest -> {
+            if (httpRequest.getHeaders().stream()
+                .noneMatch(header -> header.getName().equalsIgnoreCase("Content-Length"))) {
+                httpRequest.setHeader("Content-Length", String.valueOf(contentLength));
+            }
+        });
```

Comment:
I cannot call `requestOptions.addHeader("Content-Length", String.valueOf(contentLength));` directly, as azure-core might set the header first.

Reply:
Do we need to add a `setHeader` method on `RequestOptions`?